lib/conjur/trusted_proxy_filter.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Conjur
+  # TrustedProxyFilter wraps the default Rack IP filtering. This allows
+  # the trusted proxies to be configured using the environment variable,
+  # `TRUSTED_PROXIES`.
+  #
+  # The value of `TRUSTED_PROXIES` must be a comma-separated list of IP
+  # addresses or IP address ranges using CIDR notation.
+  #
+  # Example: TRUSTED_PROXIES=4.4.4.4,192.168.100.0/24
+  class TrustedProxyFilter
+    def initialize(wrapped_filter, env: ENV, options: { disable_cache: true })
+      @wrapped_filter = wrapped_filter
+      @env = env
+      @options = options
+      @cached_trusted_proxies = nil
+    end
+
+    def call(ip)
+      return @wrapped_filter.call(ip) unless trusted_proxies
+
+      trusted_proxies.any? { |cidr| cidr.include?(ip) }
+    end
+
+    def trusted_proxies
+      @cached_trusted_proxies || load_trusted_proxies_from_env
+    end
+
+    def load_trusted_proxies_from_env
+      return nil unless env_trusted_proxies
+
+      proxy_ips = Set.new(env_trusted_proxies.split(',') + ['127.0.0.1'])
+        .collect { |cidr| IPAddr.new(cidr.strip) }
+
+      @cached_trusted_proxies = proxy_ips unless @options[:disable_cache]
+      proxy_ips
+    end
+
+    def env_trusted_proxies
+      @env['TRUSTED_PROXIES']
+    end
```

Comment:
I think I'd just inline this.

Reply:
Yeah... I wish there was a better way to configure code climate for this. Inlining was my default, but code climate didn't like that "`@env['TRUSTED_PROXIES']` is called twice."

I'll inline again and add the exception comment.